firmware/shared/src/app/app_heartbeatMonitor.h

```diff
@@ -1,86 +1,22 @@
 #pragma once
-
 #include <stdbool.h>
-#include <stdint.h>
-#include "app_heartbeatBoardsEnum.h"
-
-#define HEARTBEAT_MONITOR_TIMEOUT_PERIOD_MS 200U
+#include "app_heartbeatMonitorBoard.h"
 
 typedef struct
 {
-    // CONFIG SETTINGS
-
-    // self_checkin for own heartbeat
-    void (*self_checkin)(bool);
-    // Override to block heartbeat faults during tests.
-    bool block_faults;
-
-    // determines if the heartbeat monitor should check in on a board
-    bool is_watching_heartbeat_for[HEARTBEAT_BOARD_COUNT];
-
-    // gives if the heartbeat is checked in.
-    // unmonitored heartbeats should be false.
-    bool heartbeats_checked_in[HEARTBEAT_BOARD_COUNT];
-    // gives if the heartbeat is in a valid state.
-    // a flag in status is true under three conditions
-    // 1) if it has been checked in
-    // 2) it was not on our list of heartbeats to check
-    // 3) if the heartbeat_checked_in is true but the timeout has not been elapsed.
-    bool status[HEARTBEAT_BOARD_COUNT];
-
-    // HEARTBEAT CAN SIGNALS
-
-    // getters for heartbeats on the CAN table
-    bool (*getters[HEARTBEAT_BOARD_COUNT])(void);
-    // resetters on the local CAN table for other heartbeats
-    // WARNING: only pass false into this function. idk how to make a closure, so we can't make this only return false
-    void (*resetters[HEARTBEAT_BOARD_COUNT])(bool);
-
-    // FAULT SETGET
-
-    // fault broadcasters for each board's heartbeat from this board
-    void (*fault_setters[HEARTBEAT_BOARD_COUNT])(bool);
-    // fault getters for each board's heartbeat from this board
-    bool (*fault_getters[HEARTBEAT_BOARD_COUNT])(void);
+    HeartbeatMonitorBoard *boards;
+    uint8_t                board_count;
+    bool                   block_faults;
+    void (*self_heartbeater)(bool);
```

Comment:
Nit: how about `self_heartbeat` or `own_heartbeat` or `heartbeat_broadcaster` or `heartbeat_tx` or something like that? 😅